IQKeyboardManager/Categories/IQUIView+Hierarchy.m

```diff
@@ -89,35 +89,54 @@ -(UIViewController*)viewController
     return nil;
 }
 
--(UIViewController *)topMostController
+- (UIViewController *)topMostController
 {
-    NSMutableArray *controllersHierarchy = [[NSMutableArray alloc] init];
-    
-    UIViewController *topController = self.window.rootViewController;
-    
-    if (topController)
+    UIViewController *result = nil;
+    UIWindow *window = [[UIApplication sharedApplication] keyWindow];
+    if (window.windowLevel != UIWindowLevelNormal)
     {
-        [controllersHierarchy addObject:topController];
+        NSArray *windows = [[UIApplication sharedApplication] windows];
+        for(UIWindow *tmpWin in windows)
+        {
+            if (tmpWin.windowLevel == UIWindowLevelNormal)
+            {
+                window = tmpWin;
+                break;
+            }
+        }
     }
     
-    while ([topController presentedViewController]) {
-        
-        topController = [topController presentedViewController];
-        [controllersHierarchy addObject:topController];
+    id  nextResponder = nil;
+    UIViewController *appRootVC = window.rootViewController;
+    if (appRootVC.presentedViewController) {
+        nextResponder = appRootVC.presentedViewController;
```

Comment:
What if user presented too many controllers?, you are only considering the first one only